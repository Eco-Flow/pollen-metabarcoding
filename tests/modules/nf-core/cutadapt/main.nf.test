nextflow_process {

    name "Test Process CUTADAPT"
    script "modules/nf-core/cutadapt/main.nf"
    process "CUTADAPT"

    test("Should run without failures") {

        when {
            params {
                clean = false
                FW_primer = "ATGCGATACTTGGTGTGAAT"
                RV_primer = "GCATATCAATAAGCGGAGGA"
                outdir = "results"
            }
            process {
                """
                input[0] = Channel.of( [ [ "id":"ERR2537816", "single_end": false ], [ file("${projectDir}/data/trimming-input/ERR2537816_1.fastq.gz"), file("${projectDir}/data/trimming-input/ERR2537816_2.fastq.gz") ] ] )
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
              assert reads.size() == 1
              assert reads[0].size() == 2
              assert reads[0][0].size() == 2
              assert reads[0][1].size() == 2
            }
            assert snapshot(
              path("${launchDir}/${params.outdir}/cutadapt/fastqs/ERR2537816_1.trim.fastq.gz"),
              path("${launchDir}/${params.outdir}/cutadapt/fastqs/ERR2537816_2.trim.fastq.gz"),
            ).match("cutadapt_fastqs")
        }

    }

}
