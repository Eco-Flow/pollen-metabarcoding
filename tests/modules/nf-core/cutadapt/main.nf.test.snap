{
    "cutadapt_fastqs": {
        "content": [
            "ERR2537816_1.trim.fastq.gz:md5,e9c82243f237a047eb2e095ed08da2f8",
            "ERR2537816_2.trim.fastq.gz:md5,757f99993765c836965d6633bfbb5c97"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-05T11:14:03.65794141"
    }
}